### PR TITLE
fix: downgrade rudderstack so debug is not pulled in

### PR DESCRIPTION
**Pull request by david-rusnak**
Closes https://github.com/influxdata/idpe/issues/18591

Downgrades rudderstack to not pull in debug `0.7.4`. Hopefully this version is ok, in the future we will probably remove rudderstck completely since it has been deprecated.

### Checklist

Authors and Reviewer(s), please verify the following:

- [ ] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [ ] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [ ] Documentation updated or issue created (provide link to issue/PR)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [ ] Feature flagged, if applicable

### CODE REVIEW

**Review by wdoconnell — COMMENTED** [review]
@david-rusnak It looks like they may have put in a hotfix in 2.46 - would you be able to check? https://github.com/rudderlabs/rudder-sdk-js/pull/1674

The PR links to a url including the line "error when upgrading rudder sdk js to the latest version"

If that still includes the `debug` package, let's stick with 2.40.5 as you propose here.

**Comment by david-rusnak** [comment]
@wdoconnell Yeah just double checked, still brings in `0.7.4`.

**Review by wdoconnell — APPROVED** [review]
Thanks @david-rusnak. In that case let's use this version of the PR as is, since upgrading would bring in the problematic dependency.

Looks good from my perspective, but please wait for @jdstrand's approval as well. Thanks!

**Review by jdstrand — COMMENTED** [review]
This looks ok to me in general, but I thought the rudderstack was updated recently to fix a different issue with CI tests? Something about it or a dependency that an earlier rudderstack used wasn't installable?

**Comment by jdstrand** [comment]
> This looks ok to me in general, but I thought the rudderstack was updated recently to fix a different issue with CI tests? Something about it or a dependency that an earlier rudderstack used wasn't installable?

That said, this did pass tests, so I guess it's not a problem.

**Review by jdstrand — APPROVED** [review]

**Comment by david-rusnak** [comment]
@jdstrand 

Yeah `2.14` had something that broke our CI, but then upgrading it to latest broke our linting because it added new types that weren't compatible - so I had to find a version that still worked for our version of typescript

**Pull request merged**
